projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -576,6 +576,26 @@ export const yieldingLoop = (count: number, chunkSize: number, callback: (index:
 
 export const isConstructor = (ref: any) => typeof ref === 'function' && Boolean(ref.prototype) && Boolean(ref.prototype.constructor);
 
+/**
+ * Bind to the i18n manager's onResourceChange event
+ * @param destroyObj Object responsible for signaling destruction of the handling object
+ * @param context Reference to the object's this context
+ */
+export function onResourceChangeHandle(destroyObj: Subject<any> | DestroyRef, callback: (event?: CustomEvent<IResourceChangeEventArgs>) => void, context: any) {
```

Comment:
If this is always meant to be called form the constructor (directly or indirectly) you could `inject(DestroyRef)` here. And avoid passing it from every component.
You can still keep it as an optional parameter.

```ts
export function onResourceChangeHandle(callback: (event?: CustomEvent<IResourceChangeEventArgs>) => void, context: any, destroyRef? DestroyRef) {
    destroyRef ??= inject(DestroyRef);
    ...
}

Reply:
For now I don't want to force it to constructor only and just less side effects in general.

Reply:
It's not force it if you keep the optional parameter.
It's true that is less "side effects" (or magic), but it's just like takeUntilDestroyed(), which is a recommended practice.
Also, currently you're adding destroyRef or destroy$ observable to many components that didn't need it. I think that with this change you'll get predictable behavior and cleaner code all over.